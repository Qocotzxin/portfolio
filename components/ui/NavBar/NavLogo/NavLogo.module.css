.outterCircle {
    animation: dash-forward 1s linear forwards;
}

.innerCircle {
    animation: dash-backward 1.2s linear forwards;
}

.text {
    opacity: 1;
    font-size: 1.25rem;
    font-family: Oswald, serif;
    text-transform: uppercase;
}

@keyframes dash-forward {
    to {
        stroke-dashoffset: 0;
    }
}

@keyframes dash-backward {
    to {
        stroke-dashoffset: 280;
    }
}