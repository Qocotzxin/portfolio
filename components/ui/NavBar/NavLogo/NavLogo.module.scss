.NavLogo {
  &__outterCircle {
    animation: dash-forward 1s linear forwards;
  }

  &__innerCircle {
    animation: dash-backward 1.2s linear forwards;
  }

  &__text {
    opacity: 1;
    font-size: 1.25rem;
    font-family: Oswald, serif;
    text-transform: uppercase;
  }
}

@keyframes dash-forward {
  to {
    stroke-dashoffset: 0;
  }
}

@keyframes dash-backward {
  to {
    stroke-dashoffset: 280;
  }
}
