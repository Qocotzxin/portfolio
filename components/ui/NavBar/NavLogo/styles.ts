import styles from "./NavLogo.module.scss";

export const Classes = {
  base: styles.NavLogo,
  outterCircle: styles.NavLogo__outterCircle,
  innerCircle: styles.NavLogo__innerCircle,
  text: styles.NavLogo__text,
};
