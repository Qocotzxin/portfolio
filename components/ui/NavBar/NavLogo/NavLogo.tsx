import { FC } from "react";
import styles from "./NavLogo.module.css";

interface NavLogoProps {
  withAnimation?: boolean;
}

const NavLogo: FC<NavLogoProps> = ({ withAnimation }) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="48"
      height="48"
      viewBox="0 0 48 48"
      aria-labelledby="title"
      role="presentation"
    >
      <title id="title" lang="en">
        Logo
      </title>
      <linearGradient
        id="gradient"
        x1="0"
        x2="0"
        y1="0"
        y2="100%"
        gradientUnits="userSpaceOnUse"
      >
        <stop stopColor="#4FD1C5" offset="40%" />
        <stop stopColor="#4299E1" offset="70%" />
      </linearGradient>
      <circle
        cx="24"
        cy="24"
        r="20"
        data-testid="NavLogo-outter-circle"
        className={withAnimation ? styles.outterCircle : ""}
        fill="none"
        stroke="#4FD1C5"
        strokeWidth="1"
        strokeDasharray="128"
        strokeDashoffset="128"
      />
      <circle
        cx="24"
        cy="24"
        r="16"
        data-testid="NavLogo-inner-circle"
        className={withAnimation ? styles.innerCircle : ""}
        fill="none"
        stroke="#4299E1"
        strokeWidth="1"
        strokeDasharray="128"
        strokeDashoffset="128"
      />
      <text
        x="14"
        y="32"
        fill="url(#gradient)"
        data-testid="NavLogo-text"
        className={`${styles.text} ${withAnimation ? "fadeIn-regular" : ""}`}
      >
        CE
      </text>
    </svg>
  );
};

export default NavLogo;
