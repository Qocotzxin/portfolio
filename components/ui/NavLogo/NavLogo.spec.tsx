import NavLogo from "./NavLogo";
import { render, screen } from "@testing-library/react";

describe("NavLogo", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<NavLogo withAnimation />);
    expect(container).toMatchSnapshot();
  });

  it("Should have animation related classes when withAnimation is true.", () => {
    render(<NavLogo withAnimation />);

    expect(screen.getByTestId("NavLogo-outter-circle")).toHaveClass(
      "_outterCircle_1ouxi_1"
    );

    expect(screen.getByTestId("NavLogo-inner-circle")).toHaveClass(
      "_innerCircle_1ouxi_5"
    );

    expect(screen.getByTestId("NavLogo-text")).toHaveClass(
      "_textAnimation_1ouxi_16"
    );
  });

  it("Should NOT have animation related classes when withAnimation is false or undefined.", () => {
    render(<NavLogo />);

    expect(screen.queryByTestId("NavLogo-outter-circle")).not.toHaveClass(
      "_outterCircle_1ouxi_1"
    );

    expect(screen.queryByTestId("NavLogo-inner-circle")).not.toHaveClass(
      "_innerCircle_1ouxi_5"
    );

    expect(screen.queryByTestId("NavLogo-text")).not.toHaveClass(
      "_textAnimation_1ouxi_16"
    );
  });
});
