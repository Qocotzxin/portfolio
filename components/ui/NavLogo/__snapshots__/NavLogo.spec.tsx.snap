// Vitest Snapshot v1

exports[`NavLogo > Should match snapshot. 1`] = `
<div>
  <svg
    aria-labelledby="title"
    height="48"
    role="presentation"
    viewBox="0 0 48 48"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="title"
      lang="en"
    >
      Logo
    </title>
    <lineargradient
      gradientUnits="userSpaceOnUse"
      id="gradient"
      x1="0"
      x2="0"
      y1="0"
      y2="100%"
    >
      <stop
        offset="40%"
        stop-color="#4FD1C5"
      />
      <stop
        offset="70%"
        stop-color="#4299E1"
      />
    </lineargradient>
    <circle
      class="_outterCircle_1ouxi_1"
      cx="24"
      cy="24"
      data-testid="NavLogo-outter-circle"
      fill="none"
      r="20"
      stroke="#4FD1C5"
      stroke-dasharray="128"
      stroke-dashoffset="128"
      stroke-width="1"
    />
    <circle
      class="_innerCircle_1ouxi_5"
      cx="24"
      cy="24"
      data-testid="NavLogo-inner-circle"
      fill="none"
      r="16"
      stroke="#4299E1"
      stroke-dasharray="128"
      stroke-dashoffset="128"
      stroke-width="1"
    />
    <text
      class="_text_1ouxi_9 _textAnimation_1ouxi_16"
      data-testid="NavLogo-text"
      fill="url(#gradient)"
      x="15"
      y="31"
    >
      CE
    </text>
  </svg>
</div>
`;
